src/components/Modals/AddContactModal.jsx

```diff
@@ -48,59 +47,45 @@ const AddContactModal = ({ addContact, showAddContactModal, setShowAddContactMod
   const { addNotification } = useNotification();
   const [userGivenName, setUserGivenName] = useState('');
   const [userFamilyName, setUserFamilyName] = useState('');
-  const [username, setUsername] = useState('');
   const [webId, setWebId] = useState('');
   const [invalidWebId, setInvalidWebId] = useState(false);
   const [processing, setProcessing] = useState(false);
   const theme = useTheme();
   const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
 
-  const wrappedSetUsername = (value) => {
-    setUsername(value);
-    const renderedWebId = renderWebId(value);
-    setWebId(renderedWebId);
-    if (invalidWebId) setInvalidWebId(false);
-  };
-
   const clearInputFields = () => {
     setUserGivenName('');
     setUserFamilyName('');
-    setUsername('');
     setWebId('');
     setInvalidWebId(false);
   };
 
   const handleAddContact = async (event) => {
     event.preventDefault();
     setProcessing(true);
+
     const { addUserGivenName, addUserFamilyName, addWebId } = event.target.elements;
+
     const userObject = {
-      givenName: addUserGivenName.value.trim(),
-      familyName: addUserFamilyName.value.trim(),
-      webId: addWebId.value.trim()
+      webId: addWebId.value.trim(),
+      ...(addUserGivenName.value.trim() && { givenName: addUserGivenName.value.trim() }),
+      ...(addUserFamilyName.value.trim() && { familyName: addUserFamilyName.value.trim() })
```

Comment:
I think `addUserGivenName.value.trim()` and `addUserFamily.value.trim()` would still be "truthy" even without the `.trim()`.

The lines could be simplified to:

```javascript
      ...(addUserGivenName.value && { givenName: addUserGivenName.value.trim() }),
      ...(addUserFamilyName.value && { familyName: addUserFamilyName.value.trim() })
```